src/lib/src/ng-select.component.ts

```diff
@@ -321,6 +334,36 @@ export class NgSelectComponent implements OnInit, OnDestroy, ControlValueAccesso
         this.itemsList.markItem(item);
     }
 
+    private handleWriteValue() {
+        // combineLatest ensures that write value is always set after latest items are loaded
+        this._writeValueSubscription = Observable.combineLatest(this._items$, this._writeValue$).subscribe((result) => {
+            if (!this._checkWriteValue) {
+                return;
```

Comment:
why this check is needed? Since handleWriteValue is called only when you set model